analyzers/tests/SonarAnalyzer.UnitTest/AnalysisContext/SonarAnalysisContextTest.Register.cs

```diff
@@ -188,6 +190,87 @@ public void SonarCompilationStartAnalysisContext_RegistSymbol_ReportsIssue()
         startContext.RaisedDiagnostic.Should().NotBeNull().And.BeSameAs(diagnostic);
     }
 
+#if NET
+
+    [DataTestMethod]
+    [DataRow("S109", "razor")]
+    [DataRow("S103", "razor")]
+    [DataRow("S1192", "razor")]
+    [DataRow("S104", "razor")]
+    [DataRow("S113", "razor")]
+    [DataRow("S1451", "razor")]
+    [DataRow("S1147", "razor")]
+    [DataRow("S109", "cshtml")]
+    [DataRow("S103", "cshtml")]
+    [DataRow("S1192", "cshtml")]
+    [DataRow("S104", "cshtml")]
+    [DataRow("S113", "cshtml")]
+    [DataRow("S1451", "cshtml")]
+    [DataRow("S1147", "cshtml")]
+    public void DisabledRules_ForRazor_DoNotRaise(string ruleId, string extension)
+    {
+        using var scope = new EnvironmentVariableScope(false) { EnableRazorAnalysis = true };
+        new VerifierBuilder()
+            .AddAnalyzer(() => new DummyAnalyzerWithLocation(ruleId, DiagnosticDescriptorFactory.MainSourceScopeTag))
+            .WithSonarProjectConfigPath(AnalysisScaffolding.CreateSonarProjectConfig(TestContext, ProjectType.Product))
+            .AddSnippet(Snippet(extension), $"SomeFile.{extension}")
+            .VerifyNoIssueReported();
+    }
+
+    [DataTestMethod]
+    [DataRow("razor")]
+    [DataRow("cshtml")]
+    public void TestRules_ForRazor_DoNotRaise(string extension)
+    {
+        using var scope = new EnvironmentVariableScope(false) { EnableRazorAnalysis = true };
+        new VerifierBuilder()
+            .AddAnalyzer(() => new DummyAnalyzerWithLocation("DummyId", DiagnosticDescriptorFactory.TestSourceScopeTag))
+            .WithSonarProjectConfigPath(AnalysisScaffolding.CreateSonarProjectConfig(TestContext, ProjectType.Test))
+            .AddSnippet(Snippet(extension), $"SomeFile.{extension}")
+            .VerifyNoIssueReported();
+    }
+
+    [DataTestMethod]
+    [DataRow("razor")]
+    [DataRow("cshtml")]
+    public void AllScopedRules_ForRazor_Raise(string extension)
+    {
+        var keyword = extension == "razor" ? "code" : "functions";
+        using var scope = new EnvironmentVariableScope(false) { EnableRazorAnalysis = true };
+        new VerifierBuilder()
+            .AddAnalyzer(() => new DummyAnalyzerWithLocation("DummyId", DiagnosticDescriptorFactory.TestSourceScopeTag, DiagnosticDescriptorFactory.MainSourceScopeTag))
+            .WithSonarProjectConfigPath(AnalysisScaffolding.CreateSonarProjectConfig(TestContext, ProjectType.Product))
+            .AddSnippet($$"""
+                        @{{keyword}}
+                        {
+                            private int magicNumber = RaiseHere(); // Noncompliant
+                            private static int RaiseHere()
+                            {
+                                return 42;
+                            }
+                        }
+                        """,
+                        $"SomeFile.{extension}")
```

Comment:
Nitpick: You could use the `Snippet` method here as well.

```suggestion
            .AddSnippet(Snippet(extension), $"SomeFile.{extension}")
```